merino/providers/suggest/adm/backends/remotesettings.py

```diff
@@ -110,7 +124,7 @@ async def fetch(self) -> SuggestionContent:
         Raises:
             RemoteSettingsError: Failed request to Remote Settings.
         """
-        suggestions: dict[str, tuple[int, int]] = {}
+        suggestions: defaultdict[str, IndexType] = defaultdict(lambda: IndexType())
```

Comment:
Two scoops of defaultdict can eliminate two if-statements below. 
```suggestion
        suggestions: defaultdict[str, IndexType] = defaultdict(lambda: defaultdict(dict))
```